src/pause_area_map.c

```diff
@@ -53,60 +138,557 @@ struct Unk_08361A7C {
+                {.timer = 2, .colors = {RGB(14, 1, 15), RGB(25, 13, 29), RGB(27, 19, 29), RGB(29, 26, 28)}},
+                {.timer = 4, .colors = {RGB(18, 1, 13), RGB(27, 13, 28), RGB(28, 19, 28), RGB(30, 25, 27)}},
+                {.timer = 6, .colors = {RGB(22, 1, 11), RGB(29, 13, 27), RGB(30, 19, 27), RGB(31, 25, 26)}},
+                {.timer = 8, .colors = {RGB(26, 2, 9), RGB(31, 14, 27), RGB(31, 19, 26), RGB(31, 25, 25)}},
+                {.timer = 6, .colors = {RGB(22, 1, 11), RGB(29, 13, 27), RGB(30, 19, 27), RGB(31, 25, 26)}},
+                {.timer = 4, .colors = {RGB(18, 1, 13), RGB(27, 13, 28), RGB(28, 19, 28), RGB(30, 25, 27)}},
+                {.timer = 2, .colors = {RGB(14, 1, 15), RGB(25, 13, 29), RGB(27, 19, 29), RGB(29, 26, 28)}},
+                {.timer = 4, .colors = {RGB(10, 0, 17), RGB(22, 13, 30), RGB(25, 19, 29), RGB(28, 26, 29)}},
+                {.timer = 6, .colors = {RGB(7, 0, 19), RGB(20, 13, 31), RGB(24, 19, 30), RGB(27, 26, 30)}},
+            },
+};
+
+static const u16 sAreaMapRoomInfoOffset[NUM_AREA_IDS] = {
+    [AREA_TUTORIAL] = 0x0,          [AREA_RAINBOW_ROUTE] = 0x0,        [AREA_MOONLIGHT_MANSION] = 0x33, [AREA_CABBAGE_CAVERN] = 0x4d,
+    [AREA_MUSTARD_MOUNTAIN] = 0x62, [AREA_CARROT_CASTLE] = 0x7b,       [AREA_OLIVE_OCEAN] = 0x92,       [AREA_PEPPERMINT_PALACE] = 0xaf,
+    [AREA_RADISH_RUINS] = 0xce,     [AREA_CANDY_CONSTELLATION] = 0xeb, [AREA_DIMENSION_MIRROR] = 0x107,
+};
 
-// TODO when pause-menu file boundaries have been set:
-// data_22.s ends with gAreaMapBigRoomKirbyOffsets, perfect for defining data above
+static const u8 sAreaMapRoomInfoLength[NUM_AREA_IDS] = {
+    [AREA_TUTORIAL] = 0x00,         [AREA_RAINBOW_ROUTE] = 0x33,       [AREA_MOONLIGHT_MANSION] = 0x1a, [AREA_CABBAGE_CAVERN] = 0x15,
+    [AREA_MUSTARD_MOUNTAIN] = 0x19, [AREA_CARROT_CASTLE] = 0x17,       [AREA_OLIVE_OCEAN] = 0x1d,       [AREA_PEPPERMINT_PALACE] = 0x1f,
+    [AREA_RADISH_RUINS] = 0x1d,     [AREA_CANDY_CONSTELLATION] = 0x1c, [AREA_DIMENSION_MIRROR] = 0x00,
+};
 
-extern const u32* const gAreaMapBGTilesets[NUM_AREA_IDS];
+/*
+ * Screen size bounds per area in tiles.
+ * Indices: 0: minX, 1: minY, 2: maxX, 3: maxY
+ */
+static const u8 sAreaMapScreenSizes[NUM_AREA_IDS][4] = {
+    [AREA_TUTORIAL] = {0, 0, 28, 18},          [AREA_RAINBOW_ROUTE] = {0, 0, 70, 40},
+    [AREA_MOONLIGHT_MANSION] = {0, 0, 60, 40}, [AREA_CABBAGE_CAVERN] = {0, 0, 58, 30},
+    [AREA_MUSTARD_MOUNTAIN] = {0, 0, 54, 36},  [AREA_CARROT_CASTLE] = {0, 0, 62, 30},
+    [AREA_OLIVE_OCEAN] = {0, 0, 60, 28},       [AREA_PEPPERMINT_PALACE] = {0, 0, 64, 36},
+    [AREA_RADISH_RUINS] = {0, 0, 58, 26},      [AREA_CANDY_CONSTELLATION] = {0, 0, 60, 40},
+    [AREA_DIMENSION_MIRROR] = {0, 0, 28, 18},
+};
+
+/*
+ * It's expected that this stays partitioned into areas, as specified by `sAreaMapRoomInfoOffset` and `sAreaMapRoomInfoLength`.
+ */
+static const struct AreaMapRoomInfo sAreaMapRoomInfos[] = {
+#include "data/pause_menu/areamap_room_infos.h"
```

Comment:
As is mentioned in the first comment, you can just paste it here. The array is not that large for us to use this kind of solution.